switch_model/solve.py

```diff
@@ -503,8 +504,17 @@ def define_arguments(argparser):
 
     # General purpose arguments
     argparser.add_argument(
-        '--verbose', '-v', dest='verbose', default=False, action='store_true',
+        '--verbose', '-v', dest='verbose', default=False, 
+        action='store_const', const=logging.WARNING,
         help='Show information about model preparation and solution')
+    argparser.add_argument(
```

Comment:
I like the idea of using logging.warn(), logging.info() and logging.debug() and support starting with that here and moving on to eventually using that throughout our code.

However, I think the logging should be more like Python's standard behavior: Logging levels are `critical`, `error`, `warning`, `info`, `debug`. The default is `warning`. For our code, I think the `warning` level should only include things that are most likely errors, that should be reported even if the user isn't looking for extra info. I'm not sure what we would use `critical` or `error` for, since those are probably things that we currently raise errors on. I think the `info` level should provide a running stream of progress messages but not much diagnostics, and the `debug` level should provide more detailed diagnostics.

With that in mind, the `info` level would be equivalent to our current `--verbose` level, and most of our current messages would go there. A few of our messages that currently start with "Warning: ..." would be moved to the warning level. And we can begin adding new debug-level messages as we add new code. 

These levels don't really map onto the flags used in this commit. We could potentially use `--warning`, `--verbose` (info level) and `--very-verbose` (debug level), but the `--warning` flag would be hard to explain (should it really be `--show-warnings`? Should there be a `--no-warning` flag?). Another option would be `--warning`, `--verbose` and `--debug`, but that conflicts with our current debug flag. I would recommend using `--log-level {warning, info, debug}`. That would also fit nicely with the existing `--log-run` and `--logs-dir` flags.

Reply:
Just noticed a lot of this is implemented in commit e5bbfd3b4fb278566f1f81fcc4515802085bc5f4. We may be almost there, but may want to change "--logging-level" to "--log-level" to be more similar to our other flags.

Reply:
I have implemented the approach I recommended above in https://github.com/switch-model/switch/pull/115/commits/29d0332abd8210bd002b0293299ef65762de38a1, but offering `--log-level {error, warning, info, debug}`. We don't offer `--log-level critical`, since that would currently be the same as `--log-level error`. 

Note that my code creates a different logger for each model. This is because each model may have different logging settings, and it is possible for several models to be created during the same Python session (possibly even existing at the same time).

There's more work to do in the future:
- replace all our `print()` calls with `logging.info()`, `logging.warn()`, etc.
- think of a good way to log stderr and stdout to file via the logger instead of using a context handler that tee's these streams. There are a couple of problems with the current approach:
  - output from the solve_scenarios module can't be logged to a file
  - all output logged to file will only go to the most recently specified file